xrpl/utils/mptoken_metadata.py

```diff
@@ -0,0 +1,397 @@
+) -> Dict[str, Any]:
+    output = {}
+    for key, value in input_dict.items():
+        mapping = next((m for m in mappings if key in (m["long"], m["compact"])), None)
+        if mapping is None:
+            output[key] = value
+            continue
+        if mapping["long"] in input_dict and mapping["compact"] in input_dict:
+            output[key] = value
+            continue
+        output[mapping["long"]] = value
+    return output
+
+
+def decode_mptoken_metadata(input_hex: str) -> MPTokenMetadata:
+    """Decodes hex-encoded MPTokenMetadata into a JSON object.
+
+    This process performs the reverse of encoding:
+    1. Converts the hex string to its JSON string representation.
+    2. Parses the JSON string into a dictionary object.
+    3. Converts compact field names within the object to their corresponding
+       long-form equivalents.
+
+    Args:
+        input_hex: The hex encoded MPTokenMetadata string.
+
+    Returns:
+        The decoded MPTokenMetadata dictionary object with long field names.
+
+    Raises:
+        Error: If the input string is not valid hex or cannot be parsed as a
+               JSON object after decoding.
+
+    Notes:
+        This utility is the counterpart to the encoding function and handles
+        field name expansion.
+    """
+    if bool(HEX_REGEX.fullmatch(input_hex)) is False:
+        raise ValueError("MPTokenMetadata must be in hex format.")
+
+    try:
+        json_metadata = json.loads(hex_to_str(input_hex))
+    except (json.JSONDecodeError, UnicodeDecodeError, ValueError) as e:
+        raise ValueError(
+            f"MPTokenMetadata is not properly formatted as JSON - {str(e)}"
+        ) from e
+
+    if not isinstance(json_metadata, dict):
+        raise TypeError("MPTokenMetadata must be a JSON object.")
+
+    output = _expand_keys(json_metadata, MPT_META_ALL_FIELDS)
+
+    if isinstance(output.get("uris"), list):
+        output["uris"] = [
+            _expand_keys(u, MPT_META_URI_FIELDS)
+            for u in output["uris"]
+            if isinstance(u, dict)
+        ]
+    if isinstance(output.get("us"), list):
+        output["us"] = [
+            _expand_keys(u, MPT_META_URI_FIELDS)
+            for u in output["us"]
+            if isinstance(u, dict)
+        ]
+
+    return output  # type: ignore
+
+
+def _shorten_keys(
+    input_dict: Dict[str, Any], mappings: List[Dict[str, Any]]
+) -> Dict[str, Any]:
+    output = {}
+    for key, value in input_dict.items():
+        mapping = next((m for m in mappings if key in (m["long"], m["compact"])), None)
+
+        if mapping is None:
+            output[key] = value
+            continue
+
+        if mapping["long"] in input_dict and mapping["compact"] in input_dict:
+            output[key] = value
+            continue
+
+        output[mapping["compact"]] = value
+    return output
+
+
+def encode_mptoken_metadata(mptoken_metadata: MPTokenMetadata) -> str:
```

Comment:
From a documentation/communication perspective, are we going to add the encode/decode logic for the MPTMetadata in the XRPL Binary Format docs page? https://xrpl.org/docs/references/protocol/binary-format

I understand that this logic is explained in the XLS specification document. However, it needs to be presented in a more readable form to the future reader as a part of the documentation.
cc @kuan121